src/features/Account/RegisterForm.tsx

```diff
@@ -6,53 +7,60 @@ import { HiAtSymbol, HiFingerPrint, HiUser } from 'react-icons/hi';
 import { InputAdornment } from '@mui/material';
 import { useFormik } from 'formik';
 import { registerValidate } from './validate';
+import { oauthButtonStyle, oauthButtonProps } from './loginTheme';
+import { signIn } from 'next-auth/react';
+import { useRouter } from 'next/router';
+
 
 const Register = () => {
   const formik = useFormik({
     initialValues: {
-      username: '',
       email: '',
       password: '',
       cpassword: '',
+      userType: '',
     },
     validate: registerValidate,
     onSubmit: onSubmit
   });
 
+  const router = useRouter();
+  const callbackUrl = (router.query?.callbackUrl as string) || '/'
+
   async function onSubmit(values) {
-    console.log(values);
+    const options = {
+      method: 'POST',
+      headers: { 'Content-Type': 'application/json' },
+      body: JSON.stringify(values)
+    }
+
+    await fetch(`${callbackUrl}api/auth/signup`, options)
```

Comment:
pretty sure this won't be a valid route with `callbackUrl` in it

Reply:
it seemed to work fine in testing; what do you mean exactly?

Reply:
The fetch request won't do anything with the callbackUrl, as far as I can tell. But with it there, there's a chance the api route won't be recognized correctly because it has extra pages (e.g. /accounts, /rodeos) in front of it). The fetch is only to send data to signup and find out whether the signup works, so we shouldn't use the callbackUrl here